### DISCUSSION TITLE
How to test getSession-protected API endpoints

### DISCUSSION THREAD

**Discussion by willnix86** [nextauthjs/next-auth#2281] [Help]
## Question 💬

I'm trying to test my Next.js API but all the endpoints are protected by Next-Auth's getSession functionality. How is it possible to mock a session before every request during testing?

**Comment by balazsorban44** [comment]
In our internal tests, we simply mock the request to `/api/auth/session` `getSession()` does.

https://github.com/nextauthjs/next-auth/blob/0475964a0f47ed5d323d0bcb779f208515ea9782/src/client/__tests__/helpers/mocks.js#L5-L13

(Don't think `ok: true` is actually needed there.)

You could also use `jest.mock` to mock the `getSession` return value.
